utility/diff.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
```

Comment:
lets use https://github.com/benjamine/jsondiffpatch for diffs, we also need to use the change collection